addons/controllers/antrea/antreaconfig_controller.go

```diff
@@ -64,31 +65,31 @@ func (r *AntreaConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, err
 	}
 
+	annotations := antreaConfig.GetAnnotations()
+	if _, ok := annotations[constants.TKGAnnotationTemplateConfig]; ok {
+		log.Info(fmt.Sprintf("resource '%v' is a config template. Skipping reconciling", req.NamespacedName))
+		return ctrl.Result{}, nil // TODO (adduarte) this should return err not nil to use backoff algorithms.
+	}
+
 	// deep copy AntreaConfig to avoid issues if in the future other controllers where interacting with the same copy
 	antreaConfig = antreaConfig.DeepCopy()
 
-	// get the parent cluster name from owner reference
-	// if the owner reference doesn't exist, use the same name as config CRD
-	clusterNamespacedName := req.NamespacedName
-	cluster := &clusterapiv1beta1.Cluster{}
-	for _, owner := range antreaConfig.OwnerReferences {
-		if owner.Kind == constants.ClusterKind {
-			clusterNamespacedName.Name = owner.Name
-			break
-		}
-	}
+	cluster, err := cutil.GetOwnerCluster(ctx, r.Client, antreaConfig, req.Namespace, constants.AntreaDefaultRefName)
 
-	// verify that the cluster related to config is present
-	if err := r.Client.Get(ctx, clusterNamespacedName, cluster); err != nil {
-		if apierrors.IsNotFound(err) {
-			log.Info(fmt.Sprintf("Cluster '%s'not found in '%s'", clusterNamespacedName.Name, clusterNamespacedName.Namespace))
-			return ctrl.Result{}, nil
+	if err != nil {
+		if apierrors.IsNotFound(err) && cluster != nil {
+			log.Info(fmt.Sprintf("'%s/%s' is listed as owner reference but could not be found",
+				cluster.Namespace, cluster.Name))
+			return ctrl.Result{}, err
```

Comment:
The return err won't be necessary since we have enqueue function here 
https://github.com/vmware-tanzu/tanzu-framework/blob/main/addons/controllers/antrea/antreaconfig_util.go#L93
So it can also help avoid error spam in addon controller

Reply:
we want to return err, because if we return nil it won't be requed. i.e. returning ctrl.Result{}, nil means the request was reconciled and it will not be requed. returning error here ensured req goes back in the queue. If not it could take too long for the config to be requed again.

Reply:
so the reason for returning err here it to take advantage of backoff reque algorithm of an error. If we "return ctrl.Result{}, nil" won't that flag the request as reconciled (which is not the case here, we want to keep trying until we find a cluster owner.

Reply:
yes you are correct. will fix

Reply:
will change to return nil